test/test_allocator.cpp

```diff
@@ -110,3 +112,29 @@ TEST(test_allocator, realloc_failing_allocators) {
   auto failing_allocator = get_failing_allocator();
   EXPECT_EQ(nullptr, rcutils_reallocf(allocated_memory, 1024, &failing_allocator));
 }
+
+TEST(test_allocator, default_allocator_maybe_fail) {
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+
+  // Check allocate
+  rcutils_fault_injection_set_count(0);
+  EXPECT_EQ(nullptr, allocator.allocate(1u, allocator.state));
+  EXPECT_EQ(RCUTILS_FAULT_INJECTION_NEVER_FAIL, rcutils_fault_injection_get_count());
```

Comment:
The usage of `RCUTILS_FAULT_INJECTION_NEVER_FAIL` is unclear to me in this scenario, `rcutils_fault_injection_get_count` is not supposed to return `0` once the code injected failed? and staying with that value until reset?

Reply:
It decrements one more time after hitting 0 to -1 and then stays there.

Reply:
It's ok then, feel free to mark this comment as resolved